core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -230,12 +227,17 @@ private[client] class CachedEventHubsReceiver private (ehConf: EventHubsConf,
       .iterator
 
     val (result, validate) = sorted.duplicate
-    assert(validate.size == newBatchSize)
+    val (validateSize, batchSizeInBytes) = validate.map(eventData => (1, eventData.getBytes.length.toLong)).reduce{
+      (countAndSize1, countAndSize2) => (countAndSize1._1 + countAndSize2._1, countAndSize1._2 + countAndSize2._2)
+    }
+    assert(validateSize == batchCount)
 
     val elapsedTimeMs = TimeUnit.NANOSECONDS.toMillis(elapsedTimeNs)
     logInfo(s"(TID $taskId) Finished receiving for $nAndP, consumer group: ${ehConf.consumerGroup
       .getOrElse(DefaultConsumerGroup)}, batchSize: $batchSize, elapsed time: $elapsedTimeMs ms")
-
+    eventHubsReceiverListener.foreach(listener => {
+      listener.onReceiveMetric(nAndP, batchCount, batchSizeInBytes, elapsedTimeMs)
+    })
```

Comment:
You can revert all changes and add the following codes to invoke metric plugin object if it is defined.
    if (metricPlugin.isDefined) {
      val (_, batchSizeInBytes) =
        validate.map(eventData => (1, eventData.getBytes.length.toLong)).reduce {
          (countAndSize1, countAndSize2) =>
            (countAndSize1._1 + countAndSize2._1, countAndSize1._2 + countAndSize2._2)
        }
      metricPlugin.foreach(listener => {
        listener.onReceiveMetric(nAndP, batchCount, batchSizeInBytes, elapsedTimeMs)
      })
    }

Reply:
Here is updated version which addresses the issue of enumerating empty map.
    if (metricPlugin.isDefined) {
      val (validateSize, batchSizeInBytes) =
        validate.map(eventData => (1, eventData.getBytes.length.toLong)).reduce {
          (countAndSize1, countAndSize2) =>
            (countAndSize1._1 + countAndSize2._1, countAndSize1._2 + countAndSize2._2)
        }
      metricPlugin.foreach(listener => {
        listener.onReceiveMetric(nAndP, batchCount, batchSizeInBytes, elapsedTimeMs)

        assert(validateSize == batchCount)
      })
    } else {
      assert(validate.size == batchCount)
    }